ports/stm32/boards/PYBD_SF6/mpconfigboard.h

```diff
@@ -69,3 +71,44 @@
 #define MICROPY_HW_ETH_RMII_TX_EN   (pyb_pin_W8)
 #define MICROPY_HW_ETH_RMII_TXD0    (pyb_pin_W45)
 #define MICROPY_HW_ETH_RMII_TXD1    (pyb_pin_W49)
+
+// Reconfigure SPI flash for dynamic size detection.
+
+#undef MICROPY_BOARD_EARLY_INIT
+#undef MICROPY_HW_SPIFLASH_SIZE_BITS
+#undef MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2
+#undef MBOOT_SPIFLASH_BYTE_SIZE
+#undef MBOOT_SPIFLASH_LAYOUT
+#undef MBOOT_SPIFLASH_ERASE_BLOCKS_PER_PAGE
+#undef MBOOT_SPIFLASH2_BYTE_SIZE
+#undef MBOOT_SPIFLASH2_LAYOUT
+#undef MBOOT_SPIFLASH2_ERASE_BLOCKS_PER_PAGE
+
+#define MICROPY_BOARD_EARLY_INIT    board_early_init_sf6
+
+#define CHIP_PARAMS0 (spi_bdev.spiflash.chip_params)
+#define CHIP_PARAMS1 (spi_bdev2.spiflash.chip_params)
```

Comment:
I'll rename these macros to something better if we agree this is a good approach.

Reply:
OK, now renamed.

(I still need to squash the fixing commits, but otherwise this should be in its final state now.)